src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1369,16 +1287,146 @@ else if (isPostGISDataStore(dataStore)) {
                 if (error != null) {
                     error = e;
                 }
-                LOGGER.log(Level.WARNING, "Error closing transaction",e);
+                LOGGER.log(Level.WARNING, "Error closing transaction", e);
             }
         }
-        
         // finally, throw any error
         if (error != null) {
             throw error;
         }
     }
 
+    private Exception copyFeaturesHighLevelAPI(ImportData data, ImportTask task,
+            DataStoreFormat format, DataStore dataStore, Transaction transaction,
+            String featureTypeName, String uniquifiedFeatureTypeName,
+            FeatureDataConverter featureDataConverter, VectorTransformChain tx) throws IOException {
+        Exception error = null;
+        ProgressMonitor monitor = task.progress();
+        try {
+            long startTime = System.currentTimeMillis();
+            task.clearMessages();
+
+            task.setTotalToProcess(format.getFeatureCount(task.getData(), task));
+            LOGGER.info("begining import - highlevel api");
+
+            FeatureSource fs = format.getFeatureSource(data, task);
+            FeatureCollection fc = fs.getFeatures();
+
+            FeatureStore featureStore = (FeatureStore) dataStore
+                    .getFeatureSource(uniquifiedFeatureTypeName);
+            featureStore.setTransaction(transaction);
+
+            fc = new TransformingCancelableInteratingFC(fc, monitor, featureDataConverter,
+                    featureStore.getSchema(), tx, task, dataStore);
```

Comment:
Would like to follow class name conventions here; there should be some base classes to start from.
- fc this is your original featureCollection
- monitor - this could only be used once, not suitable for a reusable FeatureCollection implementation
- tx, datastore - this is already known to the fc why so many parameters?